tests/com/twitter/intellij/pants/util/PantsUtilTest.java

```diff
@@ -36,4 +41,57 @@ public void run() {
     // Make sure they are identical, meaning that no new JDK was created on the 2nd find.
     assertTrue(sdkA.get() == sdkB.get());
   }
+
+  public void testListTargetsExistingNonBuildFile() {
+    File projectDir = getProjectFolder();
+    VirtualFile virtualProjectDir = LocalFileSystem.getInstance().findFileByIoFile(projectDir);
+    Optional<VirtualFile> pantsIniFileResult = PantsUtil.findPantsIniFile(Optional.ofNullable(virtualProjectDir));
+    assertTrue(pantsIniFileResult.isPresent());
+    String pantsIniFilePath = pantsIniFileResult.get().getPath();
+    File pantsIniFile = new File(pantsIniFilePath);
+    assertTrue(pantsIniFile.exists() && !pantsIniFile.isDirectory());
+    assertTrue(!PantsUtil.isBUILDFilePath(pantsIniFilePath));
+    assertEquals(PantsUtil.listAllTargets(pantsIniFilePath),
+                 Lists.newArrayList());
+  }
+
+  public void testListTargetsNonexistentNonBuildFile() {
+    String nonexistentFilePath = "not/a/build/file/path";
+    File nonexistentFile = new File(nonexistentFilePath);
+    assertTrue(!nonexistentFile.exists());
+    assertTrue(!PantsUtil.isBUILDFilePath(nonexistentFilePath));
+    assertEquals(PantsUtil.listAllTargets(nonexistentFilePath),
+                 Lists.newArrayList());
+  }
+
+  public void testListTargetsNonexistentBuildFile() {
+    String nonexistentBuildFilePath = "not/a/real/BUILD";
+    File nonexistentBuildFile = new File(nonexistentBuildFilePath);
+    assertTrue(!nonexistentBuildFile.exists());
+    assertTrue(PantsUtil.isBUILDFilePath(nonexistentBuildFilePath));
+    boolean caught = false;
+    try {
+      PantsUtil.listAllTargets(nonexistentBuildFilePath);
+    } catch (PantsException e) {
+      caught = true;
+    }
+    assertTrue("PantsException not thrown for nonexistent BUILD file", caught);
+  }
+
+  public void testListTargetInvalidBuildFile() {
+    String projectDir = getProjectFolder().getPath();
+    Path invalidBuildFilePath = Paths.get(
```

Comment:
can probably use something like https://github.com/wisechengyi/intellij-pants-plugin/blob/540e81a0b001aac8bbdef23dd966804d60cc3f4f/testFramework/com/twitter/intellij/pants/testFramework/PantsTestUtils.java#L16 to get testData

Reply:
Gotcha, I'll try to make this cleaner.